src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Container, Paper, TextField, Button, Typography, Box, Stack } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+import { useNavigate, Link } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import type { User } from '../../types/User/types';
+
+const schema = z
+  .object({
+    id: z.string().min(1).uuid().optional(),
+    fullName: z.string().min(3, 'Full name must be at least 3 characters'),
+    email: z.string().email('Invalid email address'),
+    password: z.string().min(6, 'Password must be at least 6 characters'),
+    confirm: z.string(),
+  })
+  .refine(data => data.password === data.confirm, {
+    message: "Passwords don't match",
+    path: ['confirm'],
+  });
+
+type FormData = z.infer<typeof schema>;
+
+export const Signup: React.FC = () => {
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting, isValid },
+  } = useForm<FormData>({
+    resolver: zodResolver(schema),
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
+
+  const onSubmit = async (data: FormData) => {
+    try {
+      const raw = localStorage.getItem('users');
+      const users: User[] = raw ? JSON.parse(raw) : [];
+
+      if (users.some(user => user.email === data.email)) {
+        throw new Error('Email already registered');
+      }
+
+      const newUser: User = {
+        id: uuidv4(),
+        fullName: data.fullName,
+        email: data.email,
+        password: data.password,
+      };
+
+      users.push(newUser);
+      localStorage.setItem('users', JSON.stringify(users));
+
+      navigate('/signin');
+    } catch (err: unknown) {
+      if (err instanceof Error) {
+        alert(err.message);
+      } else {
+        alert('An unknown error occurred');
+      }
+    }
+  };
+
+  return (
+    <Container maxWidth="sm">
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          mt: 8,
+        }}
+      >
+        <Paper elevation={4} sx={{ p: 4, width: '100%' }}>
+          <Typography variant="h4" align="center" gutterBottom>
+            Create Account
+          </Typography>
+
+          <Box component="form" onSubmit={handleSubmit(onSubmit)} noValidate>
+            <Stack spacing={2}>
+              <TextField
+                label="Full Name"
+                fullWidth
+                autoFocus
+                {...register('fullName')}
+                error={!!errors.fullName}
```

Comment:
Whats the point of adding !! here: 
```suggestion
                error={errors.fullName}
```

Reply:
In the line error={!!errors.fullname}, the !! converts the errors.fullname object into a boolean. If errors.fullname contains an error (i.e., if it is truthy), it will convert to true, which is passed to the error prop of the TextField. Material-UI's TextField expects a boolean for the error prop, so when error={true}, the field will display an error state (like a red border). If there is no error, errors.fullname will be falsy (such as undefined or null), and error={false} will remove the error styling.

This ensures the "Full Name" field shows the error styling (like a red border) when there's a validation issue and no error styling when the input is valid.